### PR TITLE
Update 4.7.0: Removes Default URL argument, replaces with generated long...

### PR DESCRIPTION
**Pull request by scotthasbrouck**
Adds a series of getLongURLWithParams methods in the Branch class. This breaks previous integrations by removing the DefaultURL argument - it serves no purpose. Updated README and CHANGELOG.

### CODE REVIEW

**Comment by scotthasbrouck** [comment]
@qinweigong want to let me know what you think of this?

**Comment by scotthasbrouck** [comment]
Hey @shortstuffsushi could you test this on your iOS 6 device? All you have to do is open the share sheet and this method will be called. Thanks!

File: Branch-SDK/Branch-SDK/Branch.m [hunk 1]
```diff
@@ -962,6 +979,43 @@ - (NSString *)generateShortUrl:(NSArray *)tags andAlias:(NSString *)alias andTyp
     return shortURL;
 }
 
+- (NSString *) generateLongURLWithParams:(NSDictionary *)params andChannel:(NSString *)channel andTags:(NSArray *)tags andFeature:(NSString *)feature andStage:(NSString *)stage andAlias:(NSString *)alias {
+    NSString *longURL = BNC_LINK_URL;
```
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: Would be better to use NSMutableString?

Comment (scotthasbrouck) [inline Branch-SDK/Branch-SDK/Branch.m]: Yes! Good catch. Will make the change

File: Branch-SDK/Branch-SDK/Branch.m [hunk 2]
```diff
@@ -962,6 +979,43 @@ - (NSString *)generateShortUrl:(NSArray *)tags andAlias:(NSString *)alias andTyp
     return shortURL;
 }
 
+- (NSString *) generateLongURLWithParams:(NSDictionary *)params andChannel:(NSString *)channel andTags:(NSArray *)tags andFeature:(NSString *)feature andStage:(NSString *)stage andAlias:(NSString *)alias {
+    NSMutableString *longURL = [BNC_LINK_URL mutableCopy];
+    
+    [longURL appendString:[NSString stringWithFormat:@"/a/%@?", [BNCPreferenceHelper getAppKey]]];
+    
+    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:params options:NSJSONWritingPrettyPrinted error:nil];
+    NSString *base64EncodedParams = [jsonData base64EncodedStringWithOptions:0];
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Branch.m]: This method doesn't exist on iOS 6.

Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Branch.m]: There is a method to base 64 encode in BNCPreferenceHelper.

Comment (scotthasbrouck) [inline Branch-SDK/Branch-SDK/Branch.m]: Ah, good catch.  Thanks, I'll just use that.

**Comment by shortstuffsushi** [comment]
:+1: for iOS 6.

**Comment by qinweigong** [comment]
:+1: Looks good to me too.

**Comment by shortstuffsushi** [comment]
Hey @scotthasbrouck, two notes for you:

* I'm claiming `0.4.7` for my bug fix in #62 :)
* Since you're making breaking API changes here, this should probably bump the version to `0.5.0`, but I'll leave that open for discussion (@qinwei @aaustin)

**Comment by scotthasbrouck** [comment]
Sounds good to me :+1:

**Comment by shortstuffsushi** [comment]
@scotthasbrouck, as discussed, I have merged master into your branch. I believe the conflicts are all resolved, but please let me know if I missed anything.

**Pull request merged**